Client/Frontend/Settings/AppSettingsOptions.swift

```diff
@@ -902,8 +911,28 @@ class ChinaSyncServiceSetting: WithoutAccountSetting {
     }
 
     @objc func switchValueChanged(_ toggle: UISwitch) {
-        prefs.setObject(toggle.isOn, forKey: prefKey)
-        RustFirefoxAccounts.reconfig()
+        UnifiedTelemetry.recordEvent(category: .action, method: .tap, object: .chinaServerSwitch)
+        guard profile.rustFxA.hasAccount() else {
+            prefs.setObject(toggle.isOn, forKey: prefKey)
+            RustFirefoxAccounts.reconfig()
+            return
+        }
+
+        // Show confirmation dialog for the user to sign out of FxA
+
+        let msg = "更改此设置后，再次登录您的帐户" // "Sign-in again to your account after changing this setting"
```

Comment:
nit: Would be nice to put a comment as why we are adding Chinese string while we support localization

Reply:
I agree, and it seems like we should find a way to add this to the Strings file so we use regular L10N process. I don't think we know how to specify the string is only for Chinese translation.

Reply:
Yeah we can add a comment for translators but I am not aware of a way to specify for a particular language via the system we currently have in place.